doc/manual/src/architecture/store/store.md

```diff
@@ -89,63 +75,14 @@ As it keeps track of references, it can [garbage-collect][garbage-collection] un
-| | store                  evaluates to                         | |
-| |                             |                               | |
-| |             referenced by   V       builds                  | |
-| |  [ build input ] ---> [ build plan ] ---> [ build result ]  | |
-| |         ^                                        |          | |
-| +---------|----------------------------------------|----------+ |
-+-----------|----------------------------------------|------------+
-            |                                        |
-    file system object                          store path
-            |                                        |
-+-----------|----------------------------------------|------------+
-| operating system        +------------+             |            |
-|           '------------ |            | <-----------'            |
-|                         |    file    |                          |
-|                     ,-- |            | <-,                      |
-|                     |   +------------+   |                      |
-|          execute as |                    | read, write, execute |
-|                     |   +------------+   |                      |
-|                     '-> |  process   | --'                      |
-|                         +------------+                          |
-+-----------------------------------------------------------------+
-```
-
-There exist different types of stores, which all follow this model.
-Examples:
-- store on the local file system
-- remote store accessible via SSH
-- binary cache store accessible via HTTP
+[garbage-collection]: https://en.m.wikipedia.org/wiki/Garbage_collection_(computer_science)
 
-To make store objects accessible to processes, stores ultimately have to expose store objects through the file system.
+## Two models, abstract and concrete
 
+The Nix store layer is the heart of Nix, the cornerstone of its design.
+It comes from two basic insights: a vision for build systems in the abstract based on functional programming, and an application of the vision to conventional software for conventional operating system.
+We could just present the combination of those two in the form of the current design of the Nix store, but we believe there is value introducing them separately.
+This still describes how Nix works, so this section still serves as a spec, but it also demonstrates with Nix's authors believe is a good way to think* about Nix.
+If one tries to learn the concrete specifics before learning the abstract model, the following text might come across as a wall of details without sufficient motivation.
+Conversely, if one learns the abstract model first, many of the concrete specifics will make more sense as miscellaneous details placed in the "slots" where the abstract model expects.
+The hope is that makes the material far less daunting, and helps it make sense in the mind of the reader.
```

Comment:
I had a hard time wrapping my head around this section, until I jumped ahead to read both `abstract.md` and `concrete.md`.  Basically,

+ **the "abstract" section** introduces Nix fundamental building blocks, and explains how the build prcoess works at a high level

+ **the "concrete" section** takes the introduced terms (derivation, store object, etc.) and functional programming principles, and applies them in the context of file systems

It this is correct, wouldn't it be clearer to call them model and application?

Then the  phrase  "_concrete specifics_" here means "_the application of functional programming principles to build system operations over a file system_", and in the presence of "_abstract model_" it makes me think of `design -> implementation` .